x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/AnomalyDetectorsIndex.java

```diff
@@ -144,4 +150,12 @@ public static void createStateIndexAndAliasIfNecessary(Client client, ClusterSta
         }
     }
 
+    public static String resultsMapping() {
+        return resultsMapping(MapperService.SINGLE_MAPPING_NAME);
+    }
+
+    public static String resultsMapping(String mappingType) {
```

Comment:
In `7.x` we need to still handle 6.x indices that have `doc` instead of `_doc` as the mapping type for the config and results index.